elastic/kibana
# [Index Management] Add bulk edit data retention

Closes https://github.com/elastic/kibana/issues/165142

## Summary

This PR adds bulk edit data retention to Data streams table.

On all offerings apart from serverless security project:

https://github.com/user-attachments/assets/aa091fc1-5029-4542-b76b-8045c65dda27


On serverless security project:


https://github.com/user-attachments/assets/cbccc8bd-d3f1-4511-8134-bfb0a9e8587f





### How to test:

Testing in serverless security project:
1. Start ES with `yarn es serverless --projectType security -E data_streams.lifecycle.retention.max=200d` and Kibana with `yarn serverless-security`
2. Navigate to console and create a few data streams following [these instructions](https://www.elastic.co/guide/en/elasticsearch/reference/current/set-up-a-data-stream.html#create-component-templates)
3. Navigate to Index Management -> Data streams
4. Select multiple data streams and click on bulk edit data retention button
5. Verify that:
 - The help text displays the set max data retention with a link to cloud.
 - The form doesn't allow setting a data retention higher than the max data retention.
 - If any of the data streams has a reduced data retention period, an error banner is shown, displaying the affected data streams.


Testing in stateful and searc/oblt serverless project:
1. Start ES with `yarn es serverless` and Kibana with `yarn serverless-security`
2. Navigate to console and create a few data streams following [this instructions](https://www.elastic.co/guide/en/elasticsearch/reference/current/set-up-a-data-stream.html#create-component-templates)
3. Navigate to Index Management -> Data streams
4. Select multiple data streams and click on bulk edit data retention button
5. Verify that:
 - There is no help text displaying the max data retention and no validation for the data retention period.
 - If any of the data streams has a reduced data retention period, an error banner is shown, displaying the affected data streams.

Configure cloud locally by adding the following to `config/kibana.dev.yml` and restarting Kibana:
```
xpack.cloud.id: 'test'
xpack.cloud.base_url: "https://cloud.elastic.co"
xpack.cloud.deployment_url: "/deployments/deploymentId"
```


### Checklist

- [x] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed - https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/7619
